### PR TITLE
Polyfit sensitivity

### PR DESCRIPTION
**Pull request by chiarabellenghi**
Changes to solve issue #62

### CODE REVIEW

File: skyllh/core/analysis_utils.py [hunk 1]
```diff
@@ -290,18 +290,56 @@ def estimate_mean_nsignal_for_ts_quantile(
                 p_val_weights.append(1./p1_sigma)
 
                 scanned_range = np.max(n_sig) - np.min(n_sig)
```
Comment (martwo) [inline skyllh/core/analysis_utils.py:292]: scanned_range should be moved into the next block. It is only needed there.

File: skyllh/core/analysis_utils.py [hunk 2]
```diff
@@ -109,18 +109,27 @@ def polynomial_fit(
...
-    if(deg == 2):
+    elif(deg == 2):
         (a, b, c) = (params[0], params[1], params[2])
         ns = (- b + np.sqrt((b**2)-4*a*(c-p_thr))) / (2*a)
 
         return ns
+    
+    else:
+        raise ValueError(
+                'deg = %g is not valid. The order of the polynomial function '
```
Comment (martwo) [inline skyllh/core/analysis_utils.py:130]: To much indentation

File: skyllh/core/analysis_utils.py [hunk 3]
```diff
@@ -109,18 +109,27 @@ def polynomial_fit(
     ns : float
     """
     (params, cov) = np.polyfit(ns, p, deg, w=p_weight, cov=True)
+    if(deg == 2 and params[0] > 0):
```
Comment (martwo) [inline skyllh/core/analysis_utils.py:112]: Please add a comment what this case is. It's not clear to me why you need to check for this

File: skyllh/core/analysis_utils.py [hunk 4]
```diff
@@ -109,18 +109,27 @@ def polynomial_fit(
...
+        deg = 1
+        (params, cov) = np.polyfit(ns, p, deg, w=p_weight, cov=True)
 
     if(deg == 1):
         (a, b) = (params[0], params[1])
         ns = (p_thr - b)/a
```
Comment (martwo) [inline skyllh/core/analysis_utils.py:119]: I would remove this blank line

File: skyllh/core/analysis_utils.py [hunk 5]
```diff
@@ -109,18 +109,27 @@ def polynomial_fit(
...
 
         return ns
 
-    if(deg == 2):
+    elif(deg == 2):
         (a, b, c) = (params[0], params[1], params[2])
         ns = (- b + np.sqrt((b**2)-4*a*(c-p_thr))) / (2*a)
```
Comment (martwo) [inline skyllh/core/analysis_utils.py:125]: I would remove this blank line

File: skyllh/core/analysis_utils.py [hunk 6]
```diff
@@ -230,16 +247,17 @@ def estimate_mean_nsignal_for_ts_quantile(
...
         (ts_vals0, p0_sigma, delta_p) = ([], 2*eps_p, 0)
         while (delta_p < p0_sigma*5) and (p0_sigma > eps_p):
-            ts_vals0 = np.concatenate((
+            ts_vals0 = np.concatenate((  
                 ts_vals0, ana.do_trials(
                     rss, dn_trials, mean_n_sig=ns0, bkg_kwargs=bkg_kwargs,
                     sig_kwargs=sig_kwargs, ppbar=ppbar, tl=tl)['ts']))
             (p0, p0_sigma) = calculate_pval_from_trials(ts_vals0, c)
-
+
```
Comment (martwo) [inline skyllh/core/analysis_utils.py:258]: Please configure your editor to remove trailing whitespaces.

File: skyllh/core/analysis_utils.py [hunk 7]
```diff
@@ -230,16 +247,17 @@ def estimate_mean_nsignal_for_ts_quantile(
...
+            ts_vals0 = np.concatenate((  
                 ts_vals0, ana.do_trials(
                     rss, dn_trials, mean_n_sig=ns0, bkg_kwargs=bkg_kwargs,
                     sig_kwargs=sig_kwargs, ppbar=ppbar, tl=tl)['ts']))
             (p0, p0_sigma) = calculate_pval_from_trials(ts_vals0, c)
-
+            
             n_total_generated_trials += dn_trials
-
+
```
Comment (martwo) [inline skyllh/core/analysis_utils.py:260]: Remove whitespaces here as well.

File: skyllh/core/analysis_utils.py (hunk 4 above)
Comment (chiarabellenghi) [reply skyllh/core/analysis_utils.py:119]: which one? The one between the two 'if' statements or before 'return ns'?

**Pull request merged**